src/github_utils.py

```diff
@@ -19,48 +19,56 @@ def get_pull_request_diff():
     # Get diff content
     diff_url = pull_request.diff_url
     response = requests.get(diff_url)
-    print(response.text)
+    #print(response.text)
     return response.text
 
 
-def get_valid_lines(diff_content):
-    """Parse diff to find valid line numbers with proper initialization"""
-    valid_lines = {}
+def get_valid_hunks(diff_content):
+    """Parse diff to find valid hunk positions"""
+    hunks = []
+    current_hunk = None
     current_file = None
-    line_number = None
     
     for line in diff_content.split('\n'):
-        # Reset tracking for new files
         if line.startswith('diff --git'):
-            current_file = line.split(' b/')[-1].strip()
-            valid_lines[current_file] = set()
-            line_number = None  # Reset for new file
-            continue
```

Comment:
The comment "Problematic comments structure" is only being printed if there's an exception and the `data` attribute is not present.  The `json.dumps` formatting won't apply in cases where exception `e` has a `data` attribute. This could be misleading during debugging. It might be more helpful to always include the problematic comment structure in the error message, regardless of whether the `data` attribute exists.

```suggestion
print(f"Failed to post comments: {str(e)}")
```if hasattr(e, 'data'):
print(f"Error details: {json.dumps(e.data, indent=2)}")
print("Problematic comments structure:", json.dumps(valid_comments[:2], indent=2))